vignettes/ees-api-workflow.Rmd

```diff
@@ -0,0 +1,282 @@
+  )
+```
+
+### Indicators
+
+``` {r}
+eesyapi::get_meta(dataset_id = "7c0e9201-c7c0-ff73-bee4-304e731ec0e6") |>
+  magrittr::extract2("indicators") |>
+  dplyr::filter(
+    label %in% c("Session count")
+  )
+```
+
+## Retrieving data from a data set
+
+We'll now use the element item_ids from the meta data to build queries of the data set.
+
+### Using `query_dataset()`
+
+The recommended go-to option for retrieving data is the `query_dataset()` function. This provides
+options for attaching a pre-made json file to a query, attaching a pre-made json string or passing
+a set of IDs as parameters to create a query for you.
+
+#### Querying a data set using parameters
+
+`query_dataset()` can be given a set of parameters to build up a query for you. Behind the scenes
+it creates a json string based on the parameters provided to it by the user. This inherently limits
+the extent of the type of queries that you can perform, but it should cover most basic use cases.
+
+The parameters that you'll can provide to build up your query are:
+- `time_periods`
+- `locations`
+- `filter_items`
+- `indicators` (required)
+
+All these are optional, except for `indicators` which you must always specify. If you don't provide 
+any of `time_periods`, `locations` or `filter_items`, then you#ll receive all rows from the data 
+set (although be careful, this may take some time for larger data sets).
+
+Using the parameters we obtained from `get-meta` above, then we can create a query to return 
+session count data for only **Approved educational activity**, **Authorised** and **Unauthorised** 
+in the **Absences** category, **Nationally** for **York** in **Week 26** in **2024**, we would 
+supply the following:
+
+``` {r example-parameter-query}
+eesyapi::query_dataset(
+  dataset_id = "7c0e9201-c7c0-ff73-bee4-304e731ec0e6",
+  indicators = "bqZtT",
+  filter_items = list(
+    attendance_status = c("7SdXo"),
+    attendance_type = c("6AXrf", "0k3T5", "YdkHK"),
+    day_number = "uLQo4",
+    reason = "bBrtT"
+  ),
+  geographies = c("NAT|id|dP0Zw", "LA|ID|jfhAM"),
+  time_periods = c("2024|W26")
+)
+```
+
+Some notes on the above:
+
+- `indicators`, `time_periods`, `geographies` and `filter_items` can all be supplied as vectors - 
+`c(...)` - and will be interpreted as `filter(... %in% c(...))`.
+- `filter_items` can be supplied as a list to create more stringent cross-filter combinations, i.e. 
+in the above example rows will only be returned for the specific combinations given across the 4 
+filters.
+- `geographies` can be supplied as a data frame to allow for more complex queries such as "give me 
+all LAs in a given region". The structure for this can be seen by calling `example_geography_query()`.
+
+
+#### Querying a data set using a json file or string
+
+If you're familiar with using json and wish to just write your own query from scratch, then you can 
+use `query_dataset()` with the `json_query` parameter as follows:
+
+``` {r example-json-string-query}
```

Comment:
Guess actually if you're familiar with JSON you'd make the .json file separately and read in? Would that be a better example to read in our test_query.json?

Reply:
Yes, but that just feels more awkward if this is something that someone's working through. I've tried to keep everything explicit and working out of the box - adding a the need for a file at this point felt it would be a bit clunky.